lib/config_read.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+$LOAD_PATH.push('fluentd/lib')
+
+require 'fluent/config/v1_parser'
+require 'optparse'
+
+# Accepts a file path and prints out parsed version
+class ConfigFile
+  def initialize(argv = ARGV)
+    @argv = argv
+    prepare_option_parser
+    parse_options
+    print_configurations(@file_parse)
+  end
+
+  def prepare_option_parser
+    @parser = OptionParser.new
+    @parser.banner = "\nPrint-Config-File Tool\nUsage: #{$PROGRAM_NAME} " \
+      "path/to/file\nOutput: Parsed version of config file"
+    @parser.parse!(@argv)
+  end
+
+  def usage(message = nil)
+    puts @parser.to_s
+    puts "\nError: #{message}" if message
+    exit(false)
+  end
+
+  def parse_options
```

Comment:
You actually do 2 things in this function - 
1. option validation: 1) parse the options provided as argv and 2) validate the file 
2. parse the content of filepath (and catch exeception)

If it fails at task 1, it is good to show the usage because the user does not provide the right options.
if it fails at task 2, the user needs to know the content is not a fluentd valid configuration.

Do you think it's reasonable to separate the functionality of these 2 tasks?

Reply:
That's a good idea - I'll split the function up accordingly.